Sources/MCPToolMacrosPlugin/ToolRegistration.swift

```diff
@@ -0,0 +1,65 @@
+import Foundation
+
+/// A macro that generates input schema for a Swift struct.
+/// 
+/// This macro generates a static `inputSchema` property containing the JSON schema
+/// definition for the struct, suitable for use in MCP tool definitions.
+///
+/// Usage:
+/// ```swift
+/// @Schema
+/// struct FormatTemplateInput {
+///     @Field(description: "The type of format template", constraint: .options(["commit", "pr-title"]))
+///     let formatType: String
+/// }
+/// ```
+@attached(extension, conformances: MCP.MCPParameterParsable, names: named(inputSchema), named(parseArguments), named(_requireSchema))
+public macro Schema() = #externalMacro(
```

Comment:
Macro product exposure: Since consumers may want to depend on the macro product directly, consider adding a `.macro` product for `MCPToolMacros` in `products` (can be a follow‑up as noted in your TODO).